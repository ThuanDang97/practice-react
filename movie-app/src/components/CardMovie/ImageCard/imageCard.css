.image {
  width: 220px;
  height: 330px;
}

.wrapper {
  width: 100%;
  height: 100%;
  position: relative;
  top: 0;
  left: 0;
}

.poster {
  display: inline-block;
  width: 100%;
  height: 100%;
  outline: none;
  box-sizing: border-box;
  border: 0;
}
