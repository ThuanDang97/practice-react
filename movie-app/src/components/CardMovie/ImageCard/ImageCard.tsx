import { memo } from 'react';
import { Link } from 'react-router-dom';
import Optional from '../../Optional/Optional';
import './imageCard.css';
import noImage from '../../../assets/no-image.svg';

export type TProps = {
  id: number;
  originalTitle: string;
  posterPath: string;
  onHandleHover: (id: number) => void;
};

const ImageCard = (props: TProps): JSX.Element => {
  const { id, originalTitle, posterPath, onHandleHover } = props;

  return (
    <div className="image">
      <div className="wrapper">
        <Link to={`/movie/${id}`}>
          <img
            className="poster"
            src={
              posterPath
                ? `https://www.themoviedb.org/t/p/w220_and_h330_face/${posterPath}`
                : noImage
            }
            alt={originalTitle}
          />
        </Link>
        <Optional onHandleClick={onHandleHover} id={id} />
      </div>
    </div>
  );
};

export default memo(ImageCard);
