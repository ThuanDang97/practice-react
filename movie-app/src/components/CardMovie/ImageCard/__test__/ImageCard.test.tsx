import { screen, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { setup } from '../../../../support/Setup';
import ImageCard, { TProps } from '../ImageCard';

describe('Test Content Card component', () => {
  const defaultProps: TProps = {
    id: 1,
    originalTitle: 'title',
    posterPath: 'url_path',
    onHandleHover: jest.fn(),
  };

  const customProps: TProps = {
    id: 1,
    originalTitle: 'title',
    posterPath: '',
    onHandleHover: jest.fn(),
  };

  test('Component Image Card Movie matches DOM Snapshot', () => {
    const component = (
      <Router>
        <ImageCard {...defaultProps} />;
      </Router>
    );
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Image Card Movie should render correctly', () => {
    setup(
      <Router>
        <ImageCard {...defaultProps} />;
      </Router>,
    );

    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  test('Component Image Card Movie should render default image', () => {
    setup(
      <Router>
        <ImageCard {...customProps} />;
      </Router>,
    );

    expect(screen.getByRole('img')).toBeInTheDocument();
  });
});
