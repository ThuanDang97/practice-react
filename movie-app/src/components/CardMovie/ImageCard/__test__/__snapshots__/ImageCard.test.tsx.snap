// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Content Card component Component Image Card Movie matches DOM Snapshot 1`] = `
<div>
  <div
    class="image"
  >
    <div
      class="wrapper"
    >
      <a
        href="/movie/1"
      >
        <img
          alt="title"
          class="poster"
          src="https://www.themoviedb.org/t/p/w220_and_h330_face/url_path"
        />
      </a>
      <div
        class="option"
        data-testid="optional"
      >
        <span />
      </div>
    </div>
  </div>
  ;
</div>
`;
